sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -554,8 +562,72 @@ mod checked {
                     pt,
                 )
             }
-            TransactionKind::AuthenticatorStateUpdate(_) => {
-                todo!()
+            TransactionKind::EndOfEpochTransaction(txns) => {
+                let len = txns.len();
+                for (i, tx) in txns.into_iter().enumerate() {
+                    match tx {
+                        EndOfEpochTransactionKind::ChangeEpoch(change_epoch) => {
+                            assert_eq!(i, len - 1);
+                            advance_epoch(
```

Comment:
This makes me very uneasy, as we are creating/destroying a programmable transaction runtime multiple times. Which could lead to strange errors/interactions potentially in the future. In some ways, it feels like we will end up creating yet another execution loop, which is what programmable transactions was trying to prevent! 

I think instead, We should translate the `Vec<EndOfEpochTransactionKind>` to a single `ProgrammableTransaction` and then execute it.

Reply:
done